src/link-manager.js

```diff
@@ -0,0 +1,198 @@
+/**
+ * @fileoverview Link manager for controlling the link query.
+ */
+
+/** @const */
+genotet.linkManager = {};
+
+/**
+ * Object that stores the link views by their type.
+ */
+/** @type {!Object<!Object<!Array<{
+ *    target: genotet.View,
+ *    response: string
+ *  }>>>}
+ */
+genotet.linkManager.links = {};
+
+/** @const {!Array<string>} */
+genotet.linkManager.EXPRESSION_ACTIONS = ['addProfile'];
+
+/** @const {!Array<string>} */
+genotet.linkManager.NETWORK_ACTIONS = ['nodeClick', 'edgeClick'];
+
+/** @const {!Array<string>} */
+genotet.linkManager.BINDING_ACTIONS = ['updateTrack', 'locus'];
+
+
+/**
+ * Settings for link views.
+ * This part will be extended to user self-defined in next version.
+ * @const {!Array<!Object<{
+ *    source: string,
+ *    action: string,
+ *    target: string,
+ *    response: string
+ *  }>>}
+ * */
+genotet.linkManager.LINK_SETTINGS = [
```

Comment:
I think presets layouts and their links will be preserved in the next version. What will be added is letting user edit the links. User editing links does not conflict with what you have here.

Right now you are having a huge chunk of constant defs related to presets. Presets are not quite related to link management however. A good split would be to move constants defs of presets to a dedicated file (e.g. presets.js) and then fetch preset defs from there.